crates/forge/bin/cmd/create.rs

```diff
@@ -82,11 +95,162 @@ pub struct CreateArgs {
     retry: RetryArgs,
 }
 
+/// Data used to deploy a contract on zksync
+pub struct ZkSyncData {
+    bytecode: Vec<u8>,
+    bytecode_hash: H256,
+    factory_deps: Vec<Vec<u8>>,
+}
+
 impl CreateArgs {
     /// Executes the command to create a contract
     pub async fn run(mut self) -> Result<()> {
         // Find Project & Compile
         let project = self.opts.project()?;
+        let zksync = self.opts.compiler.zk.compile.unwrap_or_default();
+        if zksync {
+            let target_path = if let Some(ref mut path) = self.contract.path {
```

Comment:
This whole flow + helpers might also fit on a zksync/module crate as a function, that would make it cleaner for merge conflicts as well